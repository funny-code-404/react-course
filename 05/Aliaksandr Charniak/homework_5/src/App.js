import React from 'react'

import List from './components/List'
import './styles.css'

class App extends React.Component {
  state = {
    id: -1,
    input: '',
    data: [],
  }

  saveState = (state) => {
    localStorage.removeItem('items')
    localStorage.removeItem('id')
    localStorage.removeItem('input')
    localStorage.setItem('items', JSON.stringify(state.data))
    localStorage.setItem('id', JSON.stringify(state.id))
    localStorage.setItem('input', JSON.stringify(state.input))
  }

  componentDidMount() {
    if (this.state.data.length === 0) {
      const data = localStorage.getItem('items') ? JSON.parse(localStorage.getItem('items')) : []
      const id = localStorage.getItem('id') ? JSON.parse(localStorage.getItem('id')) : []
      const input = localStorage.getItem('input') ? JSON.parse(localStorage.getItem('input')) : []
      this.setState({ id, input, data })
    }
  }

  handleAdd = () => {
    const { input, id } = this.state
    this.setState(
      (prevState) => ({
        ...prevState,
        id: id + 1,
        data: [...prevState.data, { id: id + 1, value: input, isDone: false }],
      }),
      () => {
        this.saveState(this.state)
      }
    )
  }

  handleChange = (e) => {
    const { value } = e.target
    this.setState(
      (prevState) => ({
        ...prevState,
        input: value,
      }),
      () => {
        this.saveState(this.state)
      }
    )
  }

  onChangeHandle = (id, isDone) => {
    this.setState(
      (prevState) => ({
        ...prevState,
        data: prevState.data.map((el) => (el.id === +id ? { ...el, isDone: isDone } : el)),
      }),
      () => {
        this.saveState(this.state)
      }
    )
  }

  deleteItem = (id) => {
    this.setState(
      (prevState) => ({
        ...prevState,
        data: prevState.data.filter((item) => item.id !== +id),
      }),
      () => {
        this.saveState(this.state)
      }
    )
  }

  render() {
    const { input, data } = this.state
    return (
      <div className="container">
        <input value={input} onChange={this.handleChange}></input>
        <button onClick={this.handleAdd}>Добавить</button>
        <List data={data} deleteItem={this.deleteItem} isDoneChange={this.onChangeHandle} />
      </div>
    )
  }
}

export default App
