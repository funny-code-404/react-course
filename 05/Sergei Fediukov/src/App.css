.App {
  /* text-align: center; */
  max-width: 1170px;
  margin: 0 auto;
  padding: 0 15px;
}
.header ul li a {
  text-decoration: none;
  color: firebrick;
}

.nav_active {
  color: red !important;
  font-weight: bold;
}
.header ul li {
  list-style: none;
  display: inline-block;
  margin-left: 20px;
  text-decoration: none;
}
.second_nav ul {
  columns: 100px 6;
}
.second_nav ul li {
  display: block;
  padding: 5px;
}
.details li {
  font-weight: bold;
  margin-top: 5px;
}
.details span {
  font-weight: bold;
}
.loading_img {
  display: flex;
  justify-content: space-around;
}

button {
  margin: 5px;
  color: firebrick;
  border: none;
  border-radius: 5px;
  background-color: #fff;
  cursor: pointer;
  outline: none;
  border: 1px solid firebrick;
}
button:hover {
  background-color: #ffe1e1;
  border: 1px solid white;
}
