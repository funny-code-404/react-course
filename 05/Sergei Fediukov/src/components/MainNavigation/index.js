import React from 'react';
import { connect } from 'react-redux';
import { getData } from '../../redux/actions';
import { NavLink } from 'react-router-dom'


class MainNavigation extends React.Component {

    handleClick = () => {
        const { path } = this.props;
        this.props.getDataAction(path);
    }
    render() {
        const { title, path } = this.props;
        return (
            <div className='menu_link' >
                <NavLink activeClassName="nav_active" onClick={this.handleClick} exact to={path}>{title}</NavLink>
            </div>
        );
    }
}

const mapStateToProps = state => state;

const mapDispatchToProps = dispatch => ({
    getDataAction: path => dispatch(getData(path)),
});


export default connect(mapStateToProps, mapDispatchToProps)(MainNavigation);
