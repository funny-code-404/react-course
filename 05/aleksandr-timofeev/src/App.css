body{
  padding:0;
  margin: 0;;
}
h1{
  margin:0;
}
nav{
  height:50px;
  margin: 0;
  padding:0;
}
nav a{
  padding:10px;
}
.wrapper-map{
  width: 800px;
  height: 600px;
}
