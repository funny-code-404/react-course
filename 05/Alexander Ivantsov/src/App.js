import React from 'react';

import './index.css';
import Task from './components/Task';
import TaskInput from './components/TaskInput';
import { getTasksLS, saveTasksLS } from './components/SaveLocal';



class App extends React.Component {
  constructor() {
    super();

    this.state = {
      tasks: []
    }
  }

  componentDidMount() {
    const localData = getTasksLS('tasks');

    if (localData === null) {
      this.setState({ tasks: [] });
    } else {
      this.setState({ tasks: localData });
    }
  }

  componentDidUpdate() {
    const { tasks } = this.state;
    saveTasksLS('tasks', tasks);
  }

  addTask = task => {
    this.setState(prevState => {
      let { tasks } = prevState;
      tasks.push({
        id: tasks.length !== 0 ? tasks.length : 0,
        title: task,
        completed: false
      });
      return tasks;
    })
  }

  completedTask = id => {
    const index = this.state.tasks.map(task => task.id).indexOf(id);

    this.setState(prevState => {
      let { tasks } = prevState;
      tasks[index].completed = true;
      return tasks;
    })
  }

  deleteTask = id => {
    this.setState({ tasks: this.state.tasks.filter(task => task.id !== id) })
  }

  render() {
    const { tasks } = this.state;

    const completedTasks = tasks.filter(task => task.completed);
    const activeTasks = tasks.filter(task => !task.completed);

    return (
      <div className="todo">
        <h1 className="counter-tasks">Tasks left: {activeTasks.length}</h1>
        {[...activeTasks, ...completedTasks].map(task => (
          <Task 
            completedTask={() => this.completedTask(task.id)}
            deleteTask={() => this.deleteTask(task.id)}
            task={task} 
            key={task.id} 
          />
        ))}
        <TaskInput addTask={this.addTask} tasks={tasks} />
      </div>
    )
  }
}

export default App;
