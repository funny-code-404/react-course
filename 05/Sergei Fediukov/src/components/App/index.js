import Routes from '../Routes'
import { paths, titles } from '../../config';
import { NavLink } from 'react-router-dom';
import MainNavigation from '../MainNavigation';
import BigLoader from '../Loader/BigLoader';
import './App.css';

function App() {
  return (
    <div className="App">
      <BigLoader />
      <div className='menu_link' >
        <NavLink activeClassName="nav_active" exact to={paths.home}>{titles.home}</NavLink>
      </div>
      <MainNavigation
        path={paths.civilizations}
        title={titles.civilizations}
      />
      <MainNavigation
        path={paths.units}
        title={titles.units}
      />
      <MainNavigation
        path={paths.structures}
        title={titles.structures}
      />
      <MainNavigation
        path={paths.technologies}
        title={titles.technologies}
      />
      <hr />
      <Routes />

    </div>
  );
}
export default App;
