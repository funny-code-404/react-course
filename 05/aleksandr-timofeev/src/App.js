import React from 'react';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
import "./App.css";
import Map from './components/map';
import Info from './components/info';
import Contacts from './components/contacts';


class App extends React.Component {

  render(){
  return (
    <div className="App">
      <Router>
        <nav>
          <Link to='/map/0/0'>Map</Link>
          <Link to='/info'>Info</Link>
          <Link to='/contacts'>Contacts</Link>
        </nav>
        <Switch>
          <Route path='/map/:id/:uuid' exact component={ Map } />
          <Route path='/info' exact component={ Info } />
          <Route path='/contacts' exact component={ Contacts } />
        </Switch>
      </Router>
    </div>
  );
}
}

export default App;
