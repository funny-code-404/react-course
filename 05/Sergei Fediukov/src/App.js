import { Navigation, Routes } from './components'
import './App.css'
import { CivilizationsProvider } from './context/CivilizationsContext/CivilizationsProvider'
import { UnitsProvider } from './context/UnitsContext/UnitsProvider'
import { StructuresProvider } from './context/StructuresContext/StructuresProvider'
import { TechnologiesProvider } from './context/TechnologiesContext/TechnologiesProvider'

const nav = [
  {
    'name': 'Home',
    'path': '/',
    'id': '1'
  },
  {
    'name': 'Civilizations',
    'path': '/civilizations',
    'id': '2'
  },
  {
    'name': 'Units',
    'path': '/units',
    'id': '3'
  },
  {
    'name': 'Structures',
    'path': '/structures',
    'id': '4'
  },
  {
    'name': 'Technologies',
    'path': '/technologies',
    'id': '5'
  },
]

function App() {
  return (

    <div className='App'>
      <TechnologiesProvider>
        <StructuresProvider>
          <UnitsProvider>
            <CivilizationsProvider>
              <Navigation menu={nav} />
              <Routes />
            </CivilizationsProvider>
          </UnitsProvider>
        </StructuresProvider>
      </TechnologiesProvider>
    </div>
  )
}
export default App;

