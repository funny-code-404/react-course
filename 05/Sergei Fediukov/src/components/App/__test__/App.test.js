import { configure, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import App from '..'
import MainNavigation from '../../../components/MainNavigation';

configure({ adapter: new Adapter() });

describe('Should render correctly "App"', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<App />)
    })
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1)
    })
    it('component have children div with "app" class', () => {
        expect(wrapper.find('.App')).toHaveLength(1)
    })
    it('component have children div with "menu_link" class', () => {
        expect(wrapper.find('.menu_link')).toHaveLength(1)
    })
    it('component have children component "MainNavigation"', () => {
        expect(wrapper.find(MainNavigation)).toHaveLength(4)
    })

})