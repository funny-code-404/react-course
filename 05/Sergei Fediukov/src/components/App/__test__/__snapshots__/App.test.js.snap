// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly "App" to Match Snapshot 1`] = `
<div
  className="App"
>
  <Connect(BigLoader) />
  <div
    className="menu_link"
  >
    <NavLink
      activeClassName="nav_active"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
  </div>
  <Connect(MainNavigation)
    path="/civilizations"
    title="Civilizations"
  />
  <Connect(MainNavigation)
    path="/units"
    title="Units"
  />
  <Connect(MainNavigation)
    path="/structures"
    title="Structures"
  />
  <Connect(MainNavigation)
    path="/technologies"
    title="Technologies"
  />
  <hr />
  <Routes />
</div>
`;
